be2-scala/src/test/scala/ch/epfl/pop/config/RuntimeEnvironmentSuite.scala

```diff
@@ -0,0 +1,27 @@
+package ch.epfl.pop.config
+
+import ch.epfl.pop.config.RuntimeEnvironment.readServerPeers
+import ch.epfl.pop.config.RuntimeEnvironmentTestingHelper.{deleteTestConfig, testWriteToServerPeersConfig}
+import org.scalatest.funsuite.{AnyFunSuiteLike => FunSuite}
+import org.scalatest.matchers.should.Matchers.{convertToAnyShouldWrapper, equal}
+
+class RuntimeEnvironmentSuite extends FunSuite {
+  test("readServerPeers() should only return the addresses that match the regex from the schema") {
+
+    // Values to add in the file
+    val addressList = List(
+      "http://127.0.0.1/",
+      "ws://127.0.0.1:7000/client",
+      "ws://127.0.0.1:8000/client",
+      "ws://127.0.0.1:9000/client"
+    )
+
+    testWriteToServerPeersConfig(addressList)
+
+    // Verify it works as expected
+    readServerPeers() should equal(addressList.tail)
+
+    // Delete the test file
+    deleteTestConfig()
```

Comment:
When left after the test assertion, the config file will not be deleted if the test fails